crates/native-c/README

```diff
@@ -0,0 +1,5 @@
+A portable native C demangler, which should mostly have byte-for-byte identical outputs to the Rust one.
```

Comment:
It might be worth noting what our expectations on safety are (e.g., do we fuzz etc sufficiently that we feel confident there's no UB on untrusted inputs)? And maybe whether we also fail in the same cases as the Rust one.

Reply:
added